### PR TITLE
Test cleanup: Test SetUpMessage in isolation

### PR DESCRIPTION
**Pull request by islemaster**
Write good unit tests for `SectionsSetUpMessage` and `CoursesSetUpMessage`, then remove deep knowledge of those components from `TeacherHomepageTest` leaving only a shallow assertion that the component is rendered with the expected props.  This is the first of many steps @Erin007 and I are taking to make `TeacherHomepageTest` less brittle, and the inner components more reusable.

Pair: @Erin007

### CODE REVIEW

File: apps/test/unit/templates/studioHomepages/SetUpMessageTest.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const wrapper = mount(
+      <CoursesSetUpMessage isRtl={false} isTeacher={false}/>
+    );
+    assert(wrapper.containsMatchingElement(
+      <div>
+        <div>
+          Start learning
+        </div>
+        <div>
+          Browse Code.org's courses to find your next challenge.
```
Comment (islemaster) [inline apps/test/unit/templates/studioHomepages/SetUpMessageTest.js]: A bit frustrating that GitHub has trouble displaying this properly even though it's valid.  Do I need to name this test file `.jsx` instead?

Comment (Hamms) [reply apps/test/unit/templates/studioHomepages/SetUpMessageTest.js]: I know that's necessary to make my linter happy

**Comment by islemaster** [comment]
You're totally right: It probably makes sense to move `CoursesSetUpMessage` and `SectionsSetUpMessage` out to become helpers in the components that consume them, and export and test `SetUpMessage` on its own.  I'd like to do that after this pure-test change.

We're leaving the props verification in `TeacherHomepageTest` in this first step - it gets pulled out of there later when we repeat this process for `Sections` and `RecentCourses`, components that sit between `TeacherHomepage` and the `SetUpMessage`s.

File: apps/test/unit/templates/studioHomepages/SetUpMessageTest.js (hunk 1 above)
Comment (islemaster) [reply apps/test/unit/templates/studioHomepages/SetUpMessageTest.js]: Yep, that helped.  Had to make a small change to make jsx files also run as part of the unit tests.

File: apps/test/unit-tests.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ function inManifest(path) {
 }
 
 
-var testsContext = require.context("./unit", true, /\.js$/);
+var testsContext = require.context("./unit", true, /\.jsx?$/);
```
Comment (Hamms) [inline apps/test/unit-tests.js:10]: 👍

**Pull request merged**